### PR TITLE
LessonService is covered by tests

### LINKED ISSUE #912: UnitTests, cover class CharlieBackend.Business.LessonService by unit tests
Cover all class by tests.

### PR DESCRIPTION
**Pull request by BobTheSuperman**
image

### CODE REVIEW

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs [hunk 1]
```diff
@@ -723,17 +859,323 @@ public async Task IsLessonDoneAsync_NotFoundLesson_ShouldThrowException()
...
+            List<Lesson> lessons = GetLessonsList();
+
+            _mentorRepositoryMock.Setup(x => x.GetMentorByAccountIdAsync(_currentUserServiceMock.Object.AccountId)).ReturnsAsync(mentor);
+            _lessonRepositoryMock.Setup(x => x.GetLessonsForMentorAsync(studentGroupId, filter.StartDate, filter.FinishDate, mentor.Id)).ReturnsAsync(lessons);
+
+            //Act
+            var result = await _lessonService.GetLessonsForMentorAsync(filter);
+
+            //Assert
+            result.Should().BeEquivalentTo(_mapper.Map<List<LessonDto>>(lessons));
```
Comment (BohdanZaliubovskyi) [inline CharlieBackend.Api.UnitTes/LessonServiceTests.cs]: Write please in multiple strings

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs [hunk 2]
```diff
@@ -723,17 +859,323 @@ public async Task IsLessonDoneAsync_NotFoundLesson_ShouldThrowException()
...
+            //Arrange
+            var mentor = new Mentor() { Id = mentorId };
+
+            _mentorRepositoryMock.Setup(x => x.GetMentorByAccountIdAsync(_currentUserServiceMock.Object.AccountId)).ReturnsAsync(mentor);
+
+            //Act
+            var result = await _lessonService.GetLessonsForMentorAsync(null);
+
+            //Assert
+            result.Should().BeEmpty();
```
Comment (BohdanZaliubovskyi) [inline CharlieBackend.Api.UnitTes/LessonServiceTests.cs]: The same

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs [hunk 3]
```diff
@@ -723,17 +859,323 @@ public async Task IsLessonDoneAsync_NotFoundLesson_ShouldThrowException()
...
+            List<Lesson> lessons = GetLessonsList();
+
+            _studentGroupRepositoryMock.Setup(x => x.GetStudentGroupsIdsByStudentId(_currentUserId)).ReturnsAsync(studentGroupsIds);
+            _lessonRepositoryMock.Setup(x => x.GetLessonsForStudentAsync(filter.StudentGroupId, filter.StartDate, filter.FinishDate, _currentUserId)).ReturnsAsync(lessons);
+
+            //Act
+            var result = await _lessonService.GetLessonsForStudentAsync(filter);
+
+            //Assert
+            result.Data.Should().BeEquivalentTo(_mapper.Map<List<LessonDto>>(lessons));
```
Comment (BohdanZaliubovskyi) [inline CharlieBackend.Api.UnitTes/LessonServiceTests.cs]: The same

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs [hunk 4]
```diff
@@ -655,31 +729,105 @@ public async Task UpdateLessonAsync()
...
+            //Arrange
+            var anyUpdateLesson = new UpdateLessonDto { ThemeName = "new theme name" };
+
+            _lessonRepositoryMock.Setup(x => x.GetByIdAsync(_id)).ReturnsAsync(() => null);
+
+            //Act
+            var result = await _lessonService.UpdateLessonAsync(_id, anyUpdateLesson);
+
+            //Assert
+            result.Should().BeNull();
```
Comment (BohdanZaliubovskyi) [inline CharlieBackend.Api.UnitTes/LessonServiceTests.cs]: Multiple strings

**Comment by BohdanZaliubovskyi** [comment]
What is percent of line coverage in this class ?

**Comment by BobTheSuperman** [comment]
> What is percent of line coverage in this class ?

Added picture about coverage

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs [hunk 5]
```diff
@@ -655,34 +777,118 @@ public async Task UpdateLessonAsync()
...
             };
 
-            var lessonService = new LessonService(
-                unitOfWork: _unitOfWorkMock.Object,
-                mapper: _mapper,
-                currentUserService: _currentUserServiceMock.Object,
-                logger: _loggerMock.Object);
-
             _lessonRepositoryMock.Setup(x => x.GetByIdAsync(lessonId)).ReturnsAsync(lesson);
             //Act
```
Comment (BohdanZaliubovskyi) [inline CharlieBackend.Api.UnitTes/LessonServiceTests.cs:878]: We need empty string before //Act

**Review by BohdanZaliubovskyi — APPROVED** [review]

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs [hunk 6]
```diff
@@ -122,34 +151,74 @@ private void MockEntities()
...
             //Assert
-            result.Should().NotBeNull();
+            result
+                .Should()
+                .NotBeNull();
         }
-        
+
         [Fact]
         public async Task CreateLessonAsync_NonExistingMentorId_ShouldReturnNotFound()
```
Comment (BohdanZaliubovskyi) [inline CharlieBackend.Api.UnitTes/LessonServiceTests.cs:230]: ...NotFoundError

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs [hunk 7]
```diff
@@ -122,34 +151,74 @@ private void MockEntities()
...
+            var result = await _lessonService.CreateLessonAsync(createLessonDTO);
+
+            //Assert
+            result
+                .Should()
+                .NotBeNull();
+        }
+
+        [Fact]
+        public async Task CreateLesson_ValidDataWithMark_ShouldReturnLessonDto()
```
Comment (BohdanZaliubovskyi) [inline CharlieBackend.Api.UnitTes/LessonServiceTests.cs:200]: Return LessonDto or NotBeNull ?

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs (hunk 6 above)
Comment (BobTheSuperman) [reply CharlieBackend.Api.UnitTes/LessonServiceTests.cs:230]: fixed

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs (hunk 7 above)
Comment (BobTheSuperman) [reply CharlieBackend.Api.UnitTes/LessonServiceTests.cs:200]: Need return LessonDto. Fixed.

File: CharlieBackend.Api.UnitTes/LessonServiceTests.cs (hunk 5 above)
Comment (BobTheSuperman) [reply CharlieBackend.Api.UnitTes/LessonServiceTests.cs:878]: added empty string

**Review by Ssashaa — APPROVED** [review]

**Review by fllimon — APPROVED** [review]

**Pull request merged**